### DISCUSSION TITLE
Documentation for plug-ins

### DISCUSSION THREAD

**Discussion by LucaCortelli** [grblHAL/core#128] [Q&A]
Hi everyone, I was wondering if there was specific documentation to create a plug-in on grblhall. can anyone give me some advice?

**Comment by terjeio** [comment]
There are some examples [here](https://github.com/grblHAL/Templates/tree/master/my_plugin).

The two data structures that plugins typically modify are documented [here](http://svn.io-engineering.com/grblHAL/html/hal_8h.html) and [here](http://svn.io-engineering.com/grblHAL/html/core__handlers_8h.html).

**Reply by LucaCortelli** [reply]
thank you very kind!
